### PR TITLE
feat(plugins): enable plugin downloading over http, https, ftp

### PR DESCRIPTION
**Pull request by link108**
enable plugins to be downloaded over http, https, and ftp

### CODE REVIEW

**Comment by link108** [comment]
internal only, please do not merge

File: kork-plugins/src/main/java/com/netflix/spinnaker/kork/plugins/spring/PluginLoader.java [hunk 1]
```diff
@@ -133,19 +132,22 @@ private boolean checkFileExists() {
...
   /**
-   * @param path
+   * @param jarLocation
    * @return path as a URL
    */
-  private URL getUrlFromPath(Path path) {
+  private URL convertToUrl(String jarLocation) {
     try {
-      return path.toUri().toURL();
+      if (jarLocation.matches("^(https?|ftp)://.*")) {
```
Comment (ncknt) [inline kork-plugins/src/main/java/com/netflix/spinnaker/kork/plugins/spring/PluginLoader.java]: That seems wrong, prefer the parsing in URL. So something like:
```
if (jarLocation.startsWith("/") {
   return Paths.get(jarLocation).toUri().toURL()
} 
 try {
  return new URL(jarLocation);
 } catch (MalformedURLException e) {
  .. do what you want here
 }

```

Comment (ncknt) [reply kork-plugins/src/main/java/com/netflix/spinnaker/kork/plugins/spring/PluginLoader.java]: And yes, it'd only accept absolute path which I think is acceptable.

**Comment by link108** [comment]
add feature flag to enable downloading, default to false

File: kork-plugins/src/main/java/com/netflix/spinnaker/kork/plugins/spring/PluginLoader.java [hunk 2]
```diff
@@ -129,23 +128,31 @@ private boolean checkFileExists() {
...
    * @return path as a URL
    */
-  private URL getUrlFromPath(Path path) {
+  private URL convertToUrl(String jarLocation, boolean downloadingEnabled) {
     try {
-      return path.toUri().toURL();
+      if (jarLocation.startsWith("/")) {
+        return Paths.get(jarLocation).toUri().toURL();
+      } else {
+        if (downloadingEnabled) {
```
Comment (ncknt) [inline kork-plugins/src/main/java/com/netflix/spinnaker/kork/plugins/spring/PluginLoader.java]: Technically `jarLocation` could also be "file://" which is not downloading. You can move your test a bit and check on the URL's protocol.

**Comment by link108** [comment]
closing to make an OSS PR

**Pull request closed without merging**